ntbs-service/Pages/ServiceDirectory/Region.cshtml.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using MoreLinq;
+using ntbs_service.DataAccess;
+using ntbs_service.Models.Entities;
+using ntbs_service.Models.ReferenceEntities;
+using ntbs_service.Services;
+
+namespace ntbs_service.Pages.ServiceDirectory
+{
+    public class RegionModel : PageModel
+    {
+        private readonly IReferenceDataRepository _referenceDataRepository;
+
+        public RegionModel(IReferenceDataRepository referenceDataRepository)
+        {
+            _referenceDataRepository = referenceDataRepository;
+        }
+        
+        [BindProperty(SupportsGet = true)]
+        public string PhecCode { get; set; }
+
+        public IList<TBService> TbServices;
+        public PHEC Phec;
+
+        public async Task<IActionResult> OnGetAsync()
+        {
+            TbServices = await _referenceDataRepository.GetTbServicesWithCaseManagersFromPhecCodeAsync(PhecCode);
+            TbServices.ForEach(x => x.CaseManagerTbServices = x.CaseManagerTbServices
+                .Where(c => c.CaseManager.IsCaseManager)
+                .ToList());
```

Comment:
Hmm, this feels a bit awkward, been umming about this a fair bit -  and the usesite has to go through the join as well, which is awkward (and e.g. means you can't filter in the usesite easily either). Feels like the datashape is the problem - so i suggest we break it up into a dictionary of lists? so something like:
```
TbServices = await _referenceDataRepository.GetTbServicesWithCaseManagersFromPhecCodeAsync(PhecCode)
            .ToDictionary(service => service, 
                   service => service.CaseManagerTbServices.Select(join => join.CaseManager).Where(cm => cs.IsCaseManager)
             );
```
or something similar to this?